pysipp/launch.py

```diff
@@ -23,112 +24,109 @@ class TimeoutError(Exception):
     "SIPp process timeout exception"
 
 
-class PopenRunner(object):
-    """Run a sequence of SIPp agents asynchronously. If any process terminates
-    with a non-zero exit code, immediately kill all remaining processes and
-    collect std streams.
+class SIPpFailure(RuntimeError):
+    """SIPp commands failed
+    """
+
 
-    Adheres to an interface similar to `multiprocessing.pool.AsyncResult`.
+class TrioRunner(object):
+    """Run a sequence of SIPp cmds asynchronously. If any process terminates
+    with a non-zero exit code, immediately cancel all remaining processes and
+    collect std streams.
     """
 
     def __init__(
         self,
-        subprocmod=subprocess,
-        osmod=os,
-        poller=select.epoll,
     ):
-        # these could optionally be rpyc proxy objs
-        self.spm = subprocmod
-        self.osm = osmod
-        self.poller = poller()
-        # collector thread placeholder
-        self._waiter = None
         # store proc results
         self._procs = OrderedDict()
 
-    def __call__(self, cmds, block=True, rate=300, **kwargs):
-        if self._waiter and self._waiter.is_alive():
+    async def run(
+        self,
+        cmds,
+        rate=300,
+        **kwargs
+    ):
+        if self.is_alive():
             raise RuntimeError(
                 "Not all processes from a prior run have completed"
             )
         if self._procs:
             raise RuntimeError(
                 "Process results have not been cleared from previous run"
             )
-        sp = self.spm
-        os = self.osm
-        DEVNULL = open(os.devnull, "wb")
-        fds2procs = OrderedDict()
-
         # run agent commands in sequence
         for cmd in cmds:
-            log.debug('launching cmd:\n"{}"\n'.format(cmd))
-            proc = sp.Popen(shlex.split(cmd), stdout=DEVNULL, stderr=sp.PIPE)
-            fd = proc.stderr.fileno()
-            log.debug("registering fd '{}' for pid '{}'".format(fd, proc.pid))
-            fds2procs[fd] = self._procs[cmd] = proc
-            # register for stderr hangup events
-            self.poller.register(proc.stderr.fileno(), select.EPOLLHUP)
+            log.debug(
+                "launching cmd:\n\"{}\"\n".format(cmd))
+            proc = await trio.open_process(
+                shlex.split(cmd),
+                stdout=subprocess.DEVNULL,
+                stderr=subprocess.PIPE
+            )
```

Comment:
``check=False,``

I think you need to disable check in order to avoid trio.run_process() to raise [subprocess.CalledProcessError](https://docs.python.org/3.11/library/subprocess.html#subprocess.CalledProcessError)

Reply:
Ah so this will resolve most of the failures then yah?

If you've already patched this please do make a PR onto this one, ow no worries but not sure when i'll get to this next.